core-web/libs/portlets/dot-ema/src/lib/feature/store/dot-ema.store.ts

```diff
@@ -61,15 +66,29 @@ export class EditEmaStore extends ComponentStore<EditEmaState> {
     readonly editor$: Observable<DotPageApiResponse> = this.select((state) => state.editor);
 
     readonly vm$ = this.select((state) => {
-        return {
-            iframeUrl: `http://localhost:3000/${state.url}?language_id=${state.editor.viewAs.language.id}`,
-            pageTitle: state.editor.page.title,
-            dialogIframeURL: state.dialogIframeURL,
-            dialogVisible: state.dialogVisible,
-            dialogHeader: state.dialogHeader,
-            dialogIframeLoading: state.dialogIframeLoading,
-            editor: state.editor
-        };
+        return state.editor.page.identifier
+            ? {
+                  iframeUrl: `http://localhost:3000/${this.createPageURL({
+                      url: state.url,
+                      language_id: state.editor.viewAs.language.id.toString(),
+                      persona_id:
+                          state.editor.viewAs.persona?.identifier ?? DEFAULT_PERSONA.identifier
+                  })}`,
+                  pageTitle: state.editor.page.title,
+                  dialogIframeURL: state.dialogIframeURL,
+                  dialogVisible: state.dialogVisible,
+                  dialogHeader: state.dialogHeader,
+                  dialogIframeLoading: state.dialogIframeLoading,
+                  editor: state.editor,
+                  selectedPersona: state.editor.viewAs.persona ?? DEFAULT_PERSONA,
+                  apiURL: `${window.location.origin}/api/v1/page/render/${this.createPageURL({
+                      url: state.url,
+                      language_id: state.editor.viewAs.language.id.toString(),
+                      persona_id:
+                          state.editor.viewAs.persona?.identifier ?? DEFAULT_PERSONA.identifier
+                  })}`
```

Comment:
WOW... let's clean this state..

`pageTitle` and `selectedPersona`, `apiURL` comes inside the `editor` we can get this from the state with a selector and some mapping... let's try to keep this state clean and slim.

Reply:
Basically anything that is already in the state in properties I don't think we should duplicated to make it surface, is better just to use selectors.